Training/Controller/Adminhtml/Note/Save.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * Codifi_Training
+ *
+ * @copyright   Copyright (c) 2021 Codifi
+ * @author      Pavel Zelenevich <pzelenevich@codifi.me>
+ */
+
+namespace Codifi\Training\Controller\Adminhtml\Note;
+
+use Magento\Framework\App\Action\Action;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\Controller\Result\JsonFactory;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Controller\Result\Json;
+use Codifi\Training\Model\NoteRepository;
+use Codifi\Training\Model\CustomerNote;
+use Magento\Backend\Model\Auth\Session;
+use Magento\Backend\Model\Session as BackendSession;
+use Exception;
+
+class Save extends Action
+{
+    /**
+     * Json factory
+     *
+     * @var JsonFactory
+     */
+    private $jsonFactory;
+
+    /**
+     * @var NoteRepository
+     */
+    private $noteRepository;
+
+    /**
+     * @var CustomerNote
+     */
+    private $note;
+
+    /**
+     * Auth session.
+     *
+     * @var Session
+     */
+    private $authSession;
+
+    /**
+     * Backend session.
+     *
+     * @var BackendSession
+     */
+    private $backendSession;
+
+    /**
+     * Save constructor.
+     *
+     * @param Context $context
+     * @param CustomerNote $note
+     * @param NoteRepository $noteRepository
+     * @param JsonFactory $jsonFactory
+     * @param Session $authSession
+     * @param BackendSession $backendSession
+     */
+    public function __construct(
+        Context $context,
+        CustomerNote $note,
+        NoteRepository $noteRepository,
+        JsonFactory $jsonFactory,
+        Session $authSession,
+        BackendSession $backendSession
```

Comment:
You can get backend session from $context (`$context->getSession()`), don't need to use dependencies